app/controllers/keywords_controller.rb

```diff
@@ -16,19 +16,21 @@ def index
   end
 
   def create
-    parse_keywords.each do |keyword|
-      current_user.keywords.create(keyword: keyword)
+    if save_keywords
+      flash[:notice] = t('keywords.upload.success')
+    else
+      flash[:alert] = keywords_form.errors.full_messages.first
     end
     redirect_to keywords_path
   end
 
   private
 
-  def parse_keywords
-    keywords_file = params[:keywords_file]
-    ParseKeywordsService.new(keywords_file).call
-  rescue GoogleSearch::Errors::KeywordsError => e
-    flash[:alert] = e.message
-    []
+  def keywords_form
+    @keywords_form ||= KeywordsForm.new(current_user)
+  end
+
+  def save_keywords
```

Comment:
Let's order this method before the `keyword_form`

Reply:
Fixed in [`4c447f6` (#42)](https://github.com/suho/google-search-ruby/pull/42/commits/4c447f699243eec5c046c00dd38c76964fc91c8f)